app/proxy/proxy_ws_test.go

```diff
@@ -434,3 +434,51 @@ func TestProxyWsConnectMutualTLS(t *testing.T) {
 
 	runTestSendReceiveWs(t, srv, proxyController)
 }
+
+func TestProxyWsConnectWithReverseProxy(t *testing.T) {
```

Comment:
Could we add another test with MutualTLS and ReverseProxy?

Reply:
Also, both tests using `NewTestHttpProxy` require authentication. So for the sake of don't leaving test code unused, we could do one of the tests with no auth (the new one I suggested or one of the existing ones).

Reply:
Two tests in `proxy_ws_tests.go` now: `TestProxyWsConnectCustomCertWithReverseProxy` and `TestProxyWsConnectMutualTLSWithReverseProxy`. 

Both identical to their non-proxy counterpart. The former does not require basic auth, the latter does.